src/users/controller/userController.ts

```diff
@@ -1,55 +1,146 @@
+  @Post('create')
+  async create(@Body() createUserDto: CreateUserDto): Promise<User> {
+    const hashedPassword = await bcrypt.hash(createUserDto.password, +process.env.BCRYPT_KEY);
+    createUserDto.password = hashedPassword;
+    createUserDto.avatar=process.env.AVATAR_URL;
     return this.usersService.create(createUserDto);
   }
 
   @UseInterceptors(ClassSerializerInterceptor)
+  @UseGuards(AuthGuard)
   @Get()
-  async findAll(): Promise<User[]> {
+  findAll(): Promise<Array<User>> {
     return this.usersService.findAll();
   }
 
   @UseInterceptors(ClassSerializerInterceptor)
-  @Get(":id")
-  async findOne(@Param("id") id: number): Promise<User> {
-    return await this.usersService.findOne(id);
+  @Post('login')
+  async login(
+    @Body() login: {email: string, password: string},
+    @Response({passthrough: true}) response
+  ): Promise<object> {
+    const user =  await this.usersService.findOneEmail(login.email);
+    if (!user) {
+      throw new BadRequestException('invalid credentials');
+    }
+    if (!await bcrypt.compare(login.password, user.password)) {
+      throw new BadRequestException('invalid credentials');
+    }
+    const accesToken:string = await this.jwtService.signAsync({id: user.id},
+      {secret: process.env.JWT_ACCESS, expiresIn: process.env.LOGIN_ACCESS_TOKEN_TIME});
+    const refreshToken:string = await this.jwtService.signAsync({id: user.id},
+      {secret: process.env.JWT_ACCESS, expiresIn: process.env.LOGIN_REFRESH_TOKEN_TIME});
+    return  {accesToken,refreshToken}
   }
 
   @UseInterceptors(ClassSerializerInterceptor)
-  @Put(":id")
+  @UseGuards(AuthGuard)
+  @Get('me')
+   findOne(@Token() data: number): Promise<User> {
+    return  this.usersService.findOne(data)
+  }
+
+  @UseInterceptors(ClassSerializerInterceptor)
+  @Put(':id')
   update(
-    @Param("id") id: number,
+    @Param('id') id: number,
     @Body() updateUserDto: UpdateUserDto
   ): Promise<User> {
     return this.usersService.update(id, updateUserDto);
   }
 
   @UseInterceptors(ClassSerializerInterceptor)
-  @Delete(":id")
-  remove(@Param("id") id: number): Promise<User> {
+  @Delete(':id')
+  remove(@Param('id') id: number): Promise<User> {
     return this.usersService.remove(id);
   }
-}
+
+  @UseInterceptors(ClassSerializerInterceptor)
+  @Post('forgot')
+  async forgotPassword(
+    @Body() forgot: {email: string},
+  ): Promise<object> {
+    const user =  await this.usersService.findOneEmail(forgot.email);
+    const accesToken:string = await this.jwtService.signAsync({id: user.id},
+      {secret: process.env.JWT_ACCESS, expiresIn: process.env.FORGOT_ACCESS_TOKEN_TIME});
+    const refreshToken:string = await this.jwtService.signAsync({id: user.id},
+      {secret: process.env.JWT_ACCESS, expiresIn: process.env.FORGOT_REFRESH_TOKEN_TIME});
+    const link =process.env.FORGOT_LINK+accesToken;
+    await sendEmail(forgot.email, link);
+    return  {accesToken,refreshToken}
```

Comment:
Why you're returning accessToken and refreshToken to the user? It's a serious security issue